broker/tasks/cloudfront.py

```diff
@@ -332,12 +341,28 @@ def update_distribution(operation_id: str, *, operation, db, **kwargs):
     config["Aliases"] = get_aliases(service_instance)
     config["CustomErrorResponses"] = get_custom_error_responses(service_instance)
 
+    tags = service_instance.tags if service_instance.tags else []
+
+    if (
+        isinstance(service_instance, CDNDedicatedWAFServiceInstance)
+        and service_instance.dedicated_waf_web_acl_arn
+    ):
+        config["WebACLId"] = service_instance.dedicated_waf_web_acl_arn
+        tags = update_cdn_with_dedicated_waf_instance_tags(tags)
```

Comment:
won't this blow up if the service instance already has the `has_dedicated_acl` tag?

Reply:
Very good point. I updated the logic to handle that case and I added a test to catch it